src/System.Private.CoreLib/src/System/Threading/ThreadPool.Unix.cs

```diff
@@ -144,22 +154,13 @@ private static IntPtr ThreadPoolDispatchCallback(IntPtr context)
              bool executeOnlyOnce,
              bool flowExecutionContext)
         {
-            //
-            // This is just a quick-and-dirty implementation to make TaskFactory.FromAsync
-            // work for the few apps that are using it.  A proper implementation would coalesce
-            // multiple waits onto a single thread, so that fewer machine resources would be
-            // consumed.
-            //
-
-            Debug.Assert(executeOnlyOnce);
-
-            QueueUserWorkItem(_ =>
-            {
-                bool timedOut = waitObject.WaitOne((int)millisecondsTimeOutInterval);
-                callBack(state, timedOut);
-            });
-
-            return new RegisteredWaitHandle();
+            ClrThreadPool.RegisteredWait registeredHandle = new ClrThreadPool.RegisteredWait(
```

Comment:
True, the double indirection is probably not a big deal but the double allocation would be good to avoid